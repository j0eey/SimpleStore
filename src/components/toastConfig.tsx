import React from 'react';
import { BaseToast, ErrorToast, BaseToastProps } from 'react-native-toast-message';
import { colors } from '../theme/Theme';
import { fonts } from '../theme/Theme';

const toastConfig = {
  success: (props: BaseToastProps) => (
    <BaseToast
      {...props}
      style={{
        zIndex: 9999,
        elevation: 9999,
        borderLeftColor: colors.SuccessToast,
        backgroundColor: colors.SuccessToast,
        minHeight: 60,
      }}
      contentContainerStyle={{ paddingHorizontal: 15 }}
      text1NumberOfLines={3}
      text1Style={{
        fontSize: 15,
        fontFamily: fonts.semiBold,
        color: colors.background,
        flexWrap: 'wrap',
      }}
      text2Style={{
        fontSize: 12,
        fontFamily: fonts.semiBold,
        color: colors.background,
        flexWrap: 'wrap',
      }}
    />
  ),
  error: (props: BaseToastProps) => (
    <ErrorToast
      {...props}
      style={{
        zIndex: 9999,
        elevation: 9999,
        borderLeftColor: colors.ErrorToast,
        backgroundColor: colors.ErrorToast,
        minHeight: 60,
      }}
      text1NumberOfLines={3}
      text1Style={{
        fontSize: 15,
        fontFamily: fonts.semiBold,
        color: colors.background,
        flexWrap: 'wrap',
      }}
    />
  ),
  info: (props: BaseToastProps) => (
    <BaseToast
      {...props}
      style={{
        zIndex: 9999,
        elevation: 9999,
        borderLeftColor: '#2196F3', // You can use a custom color or add InfoToast to your theme
        backgroundColor: '#2196F3',
        minHeight: 60,
      }}
      contentContainerStyle={{ paddingHorizontal: 15 }}
      text1NumberOfLines={3}
      text1Style={{
        fontSize: 15,
        fontFamily: fonts.semiBold,
        color: colors.background,
        flexWrap: 'wrap',
      }}
      text2Style={{
        fontSize: 12,
        fontFamily: fonts.semiBold,
        color: colors.background,
        flexWrap: 'wrap',
      }}
    />
  ),
};

export default toastConfig;
