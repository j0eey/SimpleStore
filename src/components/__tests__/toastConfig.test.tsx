import React from 'react';
import { render } from '@testing-library/react-native';
import toastConfig from '../toastConfig';
import { BaseToastProps } from 'react-native-toast-message';

// Mock the theme imports since they come from external files
jest.mock('../../theme/Theme', () => ({
  colors: {
    SuccessToast: '#4CAF50',
    ErrorToast: '#F44336',
    background: '#FFFFFF',
  },
  fonts: {
    semiBold: 'Arial-SemiBold',
  },
}));

describe('toastConfig - Snapshots', () => {
  // Sample props to test with
  const mockProps: BaseToastProps = {
    text1: 'Test Title',
    text2: 'Test Subtitle',
  };

  describe('success toast snapshots', () => {
    it('should match snapshot with basic props', () => {
      const SuccessToast = toastConfig.success;
      const { toJSON } = render(<SuccessToast {...mockProps} />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with custom props', () => {
      const customProps: BaseToastProps = {
        text1: 'Custom Success Message',
        text2: 'Custom Success Subtitle',
      };

      const SuccessToast = toastConfig.success;
      const { toJSON } = render(<SuccessToast {...customProps} />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with only text1', () => {
      const propsWithoutText2: BaseToastProps = {
        text1: 'Success Message Only',
      };

      const SuccessToast = toastConfig.success;
      const { toJSON } = render(<SuccessToast {...propsWithoutText2} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('error toast snapshots', () => {
    it('should match snapshot with basic props', () => {
      const ErrorToast = toastConfig.error;
      const { toJSON } = render(<ErrorToast {...mockProps} />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot without text2', () => {
      const propsWithoutText2: BaseToastProps = {
        text1: 'Error Message',
      };

      const ErrorToast = toastConfig.error;
      const { toJSON } = render(<ErrorToast {...propsWithoutText2} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('info toast snapshots', () => {
    it('should match snapshot with basic props', () => {
      const InfoToast = toastConfig.info;
      const { toJSON } = render(<InfoToast {...mockProps} />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with long text', () => {
      const longTextProps: BaseToastProps = {
        text1: 'This is a very long title that should wrap to multiple lines and still be displayed correctly',
        text2: 'This is also a long subtitle for testing purposes',
      };

      const InfoToast = toastConfig.info;
      const { toJSON } = render(<InfoToast {...longTextProps} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('edge cases snapshots', () => {
    it('should match snapshot with empty props', () => {
      const emptyProps: BaseToastProps = {};
      const SuccessToast = toastConfig.success;
      const { toJSON } = render(<SuccessToast {...emptyProps} />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with undefined text props', () => {
      const undefinedTextProps: BaseToastProps = {
        text1: undefined,
        text2: undefined,
      };

      const SuccessToast = toastConfig.success;
      const { toJSON } = render(<SuccessToast {...undefinedTextProps} />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot for all toast types with same props', () => {
      const testProps: BaseToastProps = {
        text1: 'Comparison Test',
        text2: 'All toast types',
      };

      // Snapshot all three types for comparison
      const SuccessToast = toastConfig.success;
      const ErrorToast = toastConfig.error;
      const InfoToast = toastConfig.info;

      const successResult = render(<SuccessToast {...testProps} />);
      const errorResult = render(<ErrorToast {...testProps} />);
      const infoResult = render(<InfoToast {...testProps} />);

      expect(successResult.toJSON()).toMatchSnapshot('success-toast');
      expect(errorResult.toJSON()).toMatchSnapshot('error-toast');
      expect(infoResult.toJSON()).toMatchSnapshot('info-toast');
    });
  });

  // Keep these unit tests as they test behavior, not rendering
  describe('toastConfig structure - Unit Tests', () => {
    it('should have all required toast types', () => {
      expect(toastConfig).toHaveProperty('success');
      expect(toastConfig).toHaveProperty('error');
      expect(toastConfig).toHaveProperty('info');
    });

    it('should have functions as values for each toast type', () => {
      expect(typeof toastConfig.success).toBe('function');
      expect(typeof toastConfig.error).toBe('function');
      expect(typeof toastConfig.info).toBe('function');
    });

    it('should return valid React components', () => {
      const SuccessComponent = toastConfig.success(mockProps);
      const ErrorComponent = toastConfig.error(mockProps);
      const InfoComponent = toastConfig.info(mockProps);

      expect(React.isValidElement(SuccessComponent)).toBe(true);
      expect(React.isValidElement(ErrorComponent)).toBe(true);
      expect(React.isValidElement(InfoComponent)).toBe(true);
    });

    it('should not crash with empty or undefined props', () => {
      const emptyProps: BaseToastProps = {};
      const undefinedTextProps: BaseToastProps = {
        text1: undefined,
        text2: undefined,
      };

      expect(() => {
        render(<toastConfig.success {...emptyProps} />);
        render(<toastConfig.error {...undefinedTextProps} />);
        render(<toastConfig.info {...emptyProps} />);
      }).not.toThrow();
    });
  });
});