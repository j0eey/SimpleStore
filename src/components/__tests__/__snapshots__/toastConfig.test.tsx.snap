// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toastConfig - Snapshots edge cases snapshots should match snapshot for all toast types with same props: error-toast 1`] = `
<View
  testID="mock-error-toast"
>
  <Text>
    Comparison Test
  </Text>
  <Text>
    All toast types
  </Text>
</View>
`;

exports[`toastConfig - Snapshots edge cases snapshots should match snapshot for all toast types with same props: info-toast 1`] = `
<View
  testID="mock-base-toast"
>
  <Text>
    Comparison Test
  </Text>
  <Text>
    All toast types
  </Text>
</View>
`;

exports[`toastConfig - Snapshots edge cases snapshots should match snapshot for all toast types with same props: success-toast 1`] = `
<View
  testID="mock-base-toast"
>
  <Text>
    Comparison Test
  </Text>
  <Text>
    All toast types
  </Text>
</View>
`;

exports[`toastConfig - Snapshots edge cases snapshots should match snapshot with empty props 1`] = `
<View
  testID="mock-base-toast"
/>
`;

exports[`toastConfig - Snapshots edge cases snapshots should match snapshot with undefined text props 1`] = `
<View
  testID="mock-base-toast"
/>
`;

exports[`toastConfig - Snapshots error toast snapshots should match snapshot with basic props 1`] = `
<View
  testID="mock-error-toast"
>
  <Text>
    Test Title
  </Text>
  <Text>
    Test Subtitle
  </Text>
</View>
`;

exports[`toastConfig - Snapshots error toast snapshots should match snapshot without text2 1`] = `
<View
  testID="mock-error-toast"
>
  <Text>
    Error Message
  </Text>
</View>
`;

exports[`toastConfig - Snapshots info toast snapshots should match snapshot with basic props 1`] = `
<View
  testID="mock-base-toast"
>
  <Text>
    Test Title
  </Text>
  <Text>
    Test Subtitle
  </Text>
</View>
`;

exports[`toastConfig - Snapshots info toast snapshots should match snapshot with long text 1`] = `
<View
  testID="mock-base-toast"
>
  <Text>
    This is a very long title that should wrap to multiple lines and still be displayed correctly
  </Text>
  <Text>
    This is also a long subtitle for testing purposes
  </Text>
</View>
`;

exports[`toastConfig - Snapshots success toast snapshots should match snapshot with basic props 1`] = `
<View
  testID="mock-base-toast"
>
  <Text>
    Test Title
  </Text>
  <Text>
    Test Subtitle
  </Text>
</View>
`;

exports[`toastConfig - Snapshots success toast snapshots should match snapshot with custom props 1`] = `
<View
  testID="mock-base-toast"
>
  <Text>
    Custom Success Message
  </Text>
  <Text>
    Custom Success Subtitle
  </Text>
</View>
`;

exports[`toastConfig - Snapshots success toast snapshots should match snapshot with only text1 1`] = `
<View
  testID="mock-base-toast"
>
  <Text>
    Success Message Only
  </Text>
</View>
`;
